src/python/python/dict_to_xml.py

```diff
@@ -0,0 +1,615 @@
+            self.file_tabs.append(0)
+            self.file_stack.append([])
+            self.write_header(Files.GEOM, '# Geometry File')
+
+            self.file_names.append('fragments/%s-emitters.xml' % base_name)
+            self.files.append(open(os.path.join(self.directory, self.file_names[Files.EMIT]), 'w', encoding='utf-8', newline="\n"))
+            self.file_tabs.append(0)
+            self.file_stack.append([])
+            self.write_header(Files.EMIT, '# Emitters File')
+
+            self.file_names.append('fragments/%s-render.xml' % base_name)
+            self.files.append(open(os.path.join(self.directory, self.file_names[Files.CAMS]), 'w', encoding='utf-8', newline="\n"))
+            self.file_tabs.append(0)
+            self.file_stack.append([])
+            self.write_header(Files.CAMS, '# Cameras and Render Parameters File')
+
+        self.set_output_file(Files.MAIN)
+
+    def set_output_file(self, file):
+        '''
+        file                int
+
+        Switch next output to the given file index
+
+        Returns None
+        '''
+
+        self.current_file = file
+
+    def write_comment(self, comment, file=None):
+        if not file:
+            file = self.current_file
+        self.wf(file, '\n')
+        self.wf(file, '<!-- %s -->\n' % comment)
+        self.wf(file, '\n')
+
+    def write_header(self, file, comment=None):
+        self.wf(file, '<?xml version="1.0" encoding="utf-8"?>\n')
+        if comment:
+            self.write_comment(comment, file)
+
+    def open_element(self, name, attributes={}, file=None):
+        if file is not None:
+            self.set_output_file(file)
+
+        self.wf(self.current_file, '<%s' % name, self.file_tabs[self.current_file])
+
+        for (k, v) in attributes.items():
+            self.wf(self.current_file, ' %s=\"%s\"' % (k, v.replace('"', '')))
+
+        self.wf(self.current_file, '>\n')
+
+        # Indent
+        self.file_tabs[self.current_file] = self.file_tabs[self.current_file] + 1
+        self.file_stack[self.current_file].append(name)
+
+    def close_element(self, file=None):
+        if file is not None:
+            self.set_output_file(file)
+
+        # Un-indent
+        self.file_tabs[self.current_file] = self.file_tabs[self.current_file] - 1
+        name = self.file_stack[self.current_file].pop()
+
+        self.wf(self.current_file, '</%s>\n' % name, self.file_tabs[self.current_file])
+
+    def element(self, name, attributes={}, file=None):
+        if file is not None:
+            self.set_output_file(file)
+
+        self.wf(self.current_file, '<%s' % name, self.file_tabs[self.current_file])
+
+        for (k, v) in attributes.items():
+            self.wf(self.current_file, ' %s=\"%s\"' % (k, v))
+
+        self.wf(self.current_file, '/>\n')
+
+    def parameter(self, paramType, paramName, attributes={}, file=None):
```

Comment:
Make sure to use then `underscore_case` naming convention for consistency with the rest of the codebase ;)

Reply:
oops, good catch